src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -57,6 +57,44 @@
 #include "utilities/events.hpp"
 #include "utilities/preserveException.hpp"
 
+class CleanupObjectMonitorsHashtable: StackObj {
+ public:
+  bool do_entry(JavaThread*& key, ObjectMonitorsHashtable::PtrList*& list) {
```

Comment:
I don't see this being called by anything ???

Reply:
See L71 below:   _ptrs->unlink(&cleanup);  // cleanup the LinkedLists
That passes the cleanup object to ResourceHashtable::unlink()
function which calls the do_entry() function in the cleanup object.

Reply:
Ah I see (well I don't really see as this is very obscure). I would have expected `CleanupObjectMonitorsHashtable` to be a subtype of something that I could then see is used by the ResourceHashtable. I'm going to guess that there is some template usage here that ensures the passed in object has the right API even if there is no subtyping relationship. Hard to see these connections at a distance.

To be clear not a review as I don't understand our Hashtable's enough to make much sense of what is actually happening here.